GooglePlayInstant/Editor/PlayInstantQuickDeployHttpHandler.cs

```diff
@@ -0,0 +1,361 @@
+            {
+                foreach (var pair in postHeaders)
+                {
+                    form.headers.Add(pair.Key, pair.Value);
+                }
+            }
+
+            var www = new WWW(endpoint, postData, form.headers);
+            return GetResultWhenDone(www);
+        }
+
+        /// <summary>
+        /// Sends a general POST request to the provided endpoint, along with the data provided in the form and headers
+        /// parameters
+        /// </summary>
+        /// <param name="endpoint">Endpoint to which the data should be sent </param>
+        /// <param name="postForm">A dictionary representing a set of key-value paris to be added to the request body</param>
+        /// <param name="postHeaders"> A dictionary representing a set of key-value pairs to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpPostRequest(string endpoint, Dictionary<string, string> postForm,
+            Dictionary<string, string> postHeaders)
+        {
+            var form = new WWWForm();
+            if (postForm != null)
+            {
+                foreach (var pair in postForm)
+                {
+                    form.AddField(pair.Key, pair.Value);
+                }
+            }
+
+            return SendHttpPostRequest(endpoint, postForm != null ? form.data : null, postHeaders);
+        }
+
+        /// <summary>
+        /// Sends a general GET request to the specified endpoint along with specified parameters and headers
+        /// </summary>
+        /// <param name="endpoint">The endpoint to with the get request should be sent</param>
+        /// <param name="getParams"> A dictionary representing ke-value pairs that should be attached to the endpoint as GET parameters</param>
+        /// <param name="getHeaders">A dictionary representing key-value pairs that constitude the data to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpGetRequest(string endpoint, Dictionary<string, string> getParams,
+            Dictionary<string, string> getHeaders)
+        {
+            var fullEndpoint = endpoint + "?";
+            var form = new WWWForm();
+            if (getHeaders != null)
+            {
+                foreach (var pair in getHeaders)
+                {
+                    form.headers.Add(pair.Key, pair.Value);
+                }
+            }
+
+            if (getParams != null)
+            {
+                foreach (var pair in getParams)
+                {
+                    fullEndpoint += pair.Key + "=" + pair.Value + "&";
+                }
+            }
+
+            var www = new WWW(fullEndpoint, null, form.headers);
+            return GetResultWhenDone(www);
+        }
+
+
+        // Waits until the WWW request is done and returns the results.
+        private static string GetResultWhenDone(WWW www)
+        {
+            while (!www.isDone)
```

Comment:
This looks like busy waiting on the main thread. Can you use an async callback instead?